### PR TITLE
Support integration tests on M1

### PR DESCRIPTION
**Pull request by meistermeier**
There are two parts in this:
1. `SkipArm64IncompatibleConfiguration` as an on-demand extension that intercepts the (parameterized) test call and inspects the arguments. If there is something that looks like a Neo4j Docker image name, it will skip those combination for which no substitution or official image exists.
2. `Neo4jImageNameSubstitutor` as an implementation of Testcontainers' `ImageNameSubstitutor` takes care about redirecting image requests to the _neo4j-arm64-experimental_ Docker hub repository.

There are already some changes in direction of #531 in there where I could not see any hard version requirements.

General rules:
- 4.4 enterprise and community is available in the official repo and will get pulled from there
- All enterprise image requests will get skipped prior 4.4
- All community images will get pulled from _neo4j/neo4j-arm64-experimental_ prior 4.4

### CODE REVIEW

File: neo4j-migrations-core/src/test/java/ac/simons/neo4j/migrations/core/Neo4jImageNameSubstitutor.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+ */
+public class Neo4jImageNameSubstitutor extends ImageNameSubstitutor {
+
+	@Override
+	public DockerImageName apply(DockerImageName dockerImageName) {
+		// don't change ryuk and neo4j 4.4+ coordinates
+		if (!dockerImageName.getRepository().startsWith("neo4j") || (dockerImageName.getRepository().startsWith("neo4j") && dockerImageName.getVersionPart().contains("4.4"))) {
+			return dockerImageName;
+		}
+		String property = System.getProperty("os.arch", "");
```
Comment (bsideup) [inline neo4j-migrations-core/src/test/java/ac/simons/neo4j/migrations/core/Neo4jImageNameSubstitutor.java:32]: Be careful, you need to check Docker's architecture, not host's.

`DockerClientFactory.instance().getInfo().getArchitecture()` would be a much better thing to check 👍

Comment (meistermeier) [reply neo4j-migrations-core/src/test/java/ac/simons/neo4j/migrations/core/Neo4jImageNameSubstitutor.java:32]: Wohooo..great. Was looking for something like this but then picked the option with the env var :D 
Thanks a lot, Sergei.

**Comment by michael-simons** [comment]
Look and works great for me. Gave it a bit of polishing.

**Comment by michael-simons** [comment]
@all-contributors please add @meistermeier for code

**Comment by michael-simons** [comment]
@all-contributors please add @bsideup for review

**Pull request merged**